### ISSUE TITLE
Looking for beta-testers

### ISSUE THREAD

**Issue by erykjj** [erykjj/jwlmanager#169]
I am adding playlist-handling functions and need volunteers to do some testing.

So, if you export playlists (for talks) or use playlists from JW Sign Language?? and would like to help, please let me know.

**Comment by elconfa** [comment]
Hi I serve in a sign language congregation and am available to help.
Massimo Confalonieri

**Comment by erykjj** [comment]
@elconfa
Great! Thank you, Massimo!
What platform/OS are you on?

**Comment by elconfa** [comment]
Android and Windows 11

**Comment by erykjj** [comment]
@elconfa 
here is the download link:
https://drive.filen.io/d/ac0807bc-8a30-44ad-8569-64164a6cc0ea#IMrkP8UARyk9NspXgNkBNTjM54UQxJ66

For now, only the (Playlist) **export** function works. Please do as much (and as methodical) testing as you can. We need to catch any bugs/discrepancies that may creep into the db in the process.

I would also like to know if and how these export files work with the JW Sign Language, and how the process can be improved/modified to make it more useful for someone like you (in a Sign Language congregation), who frequently uses playlists.

Remember this is a beta, so keep your official backups ;-)

And thank you in advance!

**Comment by erykjj** [comment]
@elconfa were you able to donwload the beta for testing?

**Comment by elconfa** [comment]
Yes, but Windows delete immediately the file. I disabled antivirus but I
can't save

Il Gio 11 Apr 2024, 15:39 Eryk J. ***@***.***> ha scritto:

**Comment by erykjj** [comment]
> Yes, but Windows delete immediately the file. I disabled antivirus but I can't save Il Gio 11 Apr 2024, 15:39 Eryk J. ***@***.***> ha scritto:

Hm... Did you ever have issues with any other releases?

Try this: https://drive.filen.io/d/ce80c3b9-5582-4e8c-a35f-0a9824d4198a#KPclHYoiy90QX80ZuTlw1vGhDyWrpBKd

Same file, but zipped. Should get you past the browser panick :-)

**Comment by erykjj** [comment]
@elconfa let me know if that worked

**Comment by glenmwright** [comment]
I too, wasn't able to use the .exe with a message from Windows saying that the file was invalid. However, the zip file worked.

I have three playlists which I exported and deleted from JWLManager. I saved the backup file and restored into JW Library. Then using the Playlist feature in JW Library I imported the JWLManager.jwlplaylist. It imported into one playlist instead of 3. This may be a limit in JW Library importing one playlist at a time.

![image](https://github.com/erykjj/jwlmanager/assets/85457329/84bf2c63-5fcd-42ef-b686-1f388d6429e0)

I imported the playlist again into my backup file with JWLManager and it too imported into one playlist file. Importing does not restore the thumbnail images or with the name of the Playlist. However, when I reimported my saved Playlist with JWLibrary it restored into two files. 

![image](https://github.com/erykjj/jwlmanager/assets/85457329/23f318b9-fbc1-4d07-a9dc-505fb12c45b5)

**Comment by erykjj** [comment]
Hi Glen,
Thank you for taking the time to help with this :-)
____
Yes, there is a limitation that I have discovered: the .jwlplaylist file can contain only _one_ playlist or it won't even open/load into JW Library. So, I was forced to export all items into _one combined playlist_ - that is so "by design" to make it function with JW Library. So, what you observed there is the way it must be.
____
I'm not sure I follow the part about importing. I have not implemented importing playlists as yet, only opening the archive by itself works. Please confirm that this is what you have done and that's where no thumbnails were present.

**Comment by glenmwright** [comment]
Let's try again, I got lost while I was trying to explain it too. 😁

I exported my 3 playlists with JWLM and then imported that file into JW Library. There are no thumbnail images, but the playlist plays correctly. Closing JW Library and reopening still does not show the thumbnails.

![image](https://github.com/erykjj/jwlmanager/assets/85457329/b05a723f-53d7-448d-bae2-6655fd340a8d)

I wonder if there was corruption in my JW Library backup file that caused the duplicate playlist when importing through JW Library. I will keep testing.

**Comment by erykjj** [comment]
> I exported my 3 playlists with JWLM and then imported that file into JW Library. There are no thumbnail images, but the playlist plays correctly. Closing JW Library and reopening still does not show the thumbnails.

Noted.

**Comment by erykjj** [comment]
@glenmwright please try again. I just tried exporting from Manager and opening in JWL and thumbnails worked.

**Comment by erykjj** [comment]
Hi @elconfa & @glenmwright 

I have implemented the playlist **export** now as well, but it needs _thorough_ testing - especially with video items that are trimmed or used with publications (Sign Language).

https://drive.filen.io/d/7bdf239e-e4f9-489e-b71f-ed035908b1bd#7b5bI3YhcVzDcE28R1LfRTKvbhUnZcGl

Please report any strange/unexpected behavior.

Roadmap: add images to (new) playlist.

**Comment by glenmwright** [comment]
Hi Eryk, 

Sorry Windows considers the file unsafe and immediately deletes it when extracting it from the zip file.

**Comment by erykjj** [comment]
> Hi Eryk,
> 
> Sorry Windows considers the file unsafe and immediately deletes it when extracting it from the zip file.

Hm... Wow, Windows can be so strange sometimes... Thank you for reporting!

Let me try again...

**Comment by erykjj** [comment]
Please try this: https://drive.filen.io/d/d1827df9-678b-44bf-be24-9bb7c6797443#vcAPbaXyagkGCdFvLiVDI7djqTmHXHaf

It's a zip of the exe that has been scanned by virustotal.com, only producing the typical (three) false-positives: https://www.virustotal.com/gui/file/2908f8fcf1443d6add504f5a25c2ceea8f9656e06967b74ae03ecbf2b3740687/behavior

**Comment by glenmwright** [comment]
Yes, Beta 2 is working.

**Comment by erykjj** [comment]
I have implemented an image-adding (to playlist) interface.
Please test and report as you are able. Thank you!

https://drive.filen.io/d/732a6091-ba74-4d96-b60f-aa5ad9be692c#jhGw9BaOUJzAOVMPa4H9JOfTPUKKuydu

**Comment by erykjj** [comment]
Here is beta3: https://drive.filen.io/d/2b1bc1e8-1012-4d1d-a4af-cd4c9c35a643#spdRTCkoqz1LMnxPMGvnHbyQP2h0jEd3

**Comment by erykjj** [comment]
@elconfa & @jonharrell (& anyone else):

I believe the Playlist functions are fairly stable, BUT I would appreciate some more eyes to do some more testings:

[JWLManager_v4.5.0-RC](https://drive.filen.io/d/ceefdfdf-9de8-4e2c-98e1-613b606f48f0#Jw8PZEpBFnJB34mRap92KAat3ULc7dmS)

Does it behave like you would expect it to behave? Any strange/odd/unusual results? Is some functionality missing that would be useful?

**Comment by jonharrell** [comment]
Thank you for your hard work! I'm sorry I got sick and then buried with
projects... This is specific project is close to my day to day needs, I'll
jump on testing this right away

Jon Harrell

**Comment by erykjj** [comment]
> Thank you for your hard work! I'm sorry I got sick and then buried with projects... This is specific project is close to my day to day needs, I'll jump on testing this right away Jon Harrell

Glad to hear you are better :-)
Thanks in advance, @jonharrell !

**Comment by elconfa** [comment]
Hi, I tested beta 3 but with JwLibrary sign language it doesn't work well. With the sign language app we only use playlists. After importing two playlists and saving the file to import it into Jwlibrary, the 2 playlists are messed up with various errors and links to paragraphs that lead to biblical verses. I'll try the new beta and let you know. THANK YOU

**Comment by erykjj** [comment]
> Hi, I tested beta 3 but with JwLibrary sign language it doesn't work well. With the sign language app we only use playlists. After importing two playlists and saving the file to import it into Jwlibrary, the 2 playlists are messed up with various errors and links to paragraphs that lead to biblical verses. I'll try the new beta and let you know. THANK YOU

Hi @elconfa 
Thank you for reporting. Yes, please try with the RC. If you still get the errors, please send me the two files you are importing so I can try to track down the bugs.

**Comment by elconfa** [comment]
> > Hi, I tested beta 3 but with JwLibrary sign language it doesn't work well. With the sign language app we only use playlists. After importing two playlists and saving the file to import it into Jwlibrary, the 2 playlists are messed up with various errors and links to paragraphs that lead to biblical verses. I'll try the new beta and let you know. THANK YOU
> 
> Hi @elconfa Thank you for reporting. Yes, please try with the RC. If you still get the errors, please send me the two files you are importing so I can try to track down the bugs.

[2 playlists.zip](https://github.com/erykjj/jwlmanager/files/15220025/2.playlists.zip)

Same problem.
Please try this 2 export playlist

**Comment by erykjj** [comment]
@elconfa
The two files you sent me are both .jwlibrary archives, not playlists (.jwlplaylist). What is it that you do with these that messes up?

I understood you imported two _playlists_ into JWLManager and exported selected items as a new playlist, but that's obviously not the case here. Please explain again :-)

Eryk

**Comment by erykjj** [comment]
OK, so those two files are full backups of the JW Sign app.
And you cannot select only one playlist to share? It's all or nothing?
Now, precisely, what would you like to be able to do with that full backup?

**Comment by elconfa** [comment]
I'll explain the problem: for the midweek meeting the brother in charge of conducting the congregation study creates his own playlist (full backup), while the brother in charge of video direction prepares a playlist (full backup) that includes all the others speeches.
The 2 playlists are exported from different PCs, but cannot be imported into the Kingdom Hall PC at the same time.
In practice, the brother in charge of the video department must upload the playlist of the speeches, then when the congregational study begins he must import the new playlist and immediately afterwards re-import the first one for the final announcements and the song.
I was hoping that this project could combine the 2 playlists together and allow importing both at the same time.
I hope I explained myself.
Thank you

**Comment by erykjj** [comment]
Got it!
It does sound rather awkward to have to do it like that...
Yes, JWLManager will be able to handle that. I will think about simplifying it even more so you don't have to export and re-import ;-)

**Comment by erykjj** [comment]
So, just to confirm, if you were able to **combine both lists in one .jwlplaylist** file (one playlist) to import into JW Library on the KH PC that would be satisfactory, correct?

**Comment by jonharrell** [comment]
That would be great. Also if possible adding one playlist from a backup or
removing them one by one would allow custom merging of full backups.

Jon Harrell

**Comment by erykjj** [comment]
@jonharrell & @elconfa
 
OK, please test again with RC2: https://drive.filen.io/d/491d4f69-b525-438f-a342-5d2945735024#Sf6mxKNQOAn7aO7PTBJC4aNZjwBJxrlg

You can import .jwlibrary (full backup) into playlists using the Import button (don't drag-and-drop, as it will open that archive instead of importing the playlists). Then try exporting selected items into a new playlist to see if it does/behaves as expected.

**Comment by elconfa** [comment]
Hi, I opened playlist 1 and then imported playlist 2. Saved it in a new file and imported it into the JW application.
This works. I'll do some testing in the next few days.
Thank you so much

**Comment by erykjj** [comment]
> Hi, I opened playlist 1 and then imported playlist 2. Saved it in a new file and imported it into the JW application. This works. I'll do some testing in the next few days. Thank you so much

Yes, please test as intensively as possible.

**Comment by jonharrell** [comment]
So far it is looking good. I tried importing some old ASL talks as well as
BXE Bulgarian cyrillic titled talks and personal playlists. I also removed
a single talk playlist and rotated through various combinations and it
seems to be working good. Some UI choices are not as obvious at first but
the program is working.

Jonathan Harrell
***@***.***

**Comment by erykjj** [comment]
> So far it is looking good. I tried importing some old ASL talks as well as BXE Bulgarian cyrillic titled talks and personal playlists. I also removed a single talk playlist and rotated through various combinations and it seems to be working good. Some UI choices are not as obvious at first but the program is working. Jonathan Harrell

@jonharrell feel free to suggest UI adjustments ;-)

**Comment by erykjj** [comment]
@jonharrell & @elconfa
 
Just wondering how your testing is coming along? Any observations to report?

**Comment by erykjj** [comment]
I guess I will take this to mean there are no issues to report ;-)

**Comment by erykjj** [comment]
Closing with release of [v4.5.0](https://github.com/erykjj/jwlmanager/releases/tag/v4.5.0)
